src/libostree/ostree-repo-pull.c

```diff
@@ -2413,6 +2413,13 @@ ostree_repo_pull_with_options (OstreeRepo             *self,
         }
     }
 
+  /* For local pulls, default to disabling static deltas so that the
+   * exact object files are copied.
+   */
+  if (pull_data->remote_repo_local &&
+      !(disable_static_deltas || require_static_deltas))
```

Comment:
Out of curiosity, why not just `if (pull_data->remote_repo_local && !require_static_deltas)` ?